src/App.js

```diff
@@ -70,89 +70,78 @@ const Transition = styled(CSSTransition)`
-          userId: userInfo.userId,
-          token: userInfo.token,
-        });
-      } else {
-        console.log("user info not found");
-      }
-    } catch {
-      console.log("Cannot find info from localStorage");
+const loadUserInfo = (props) => {
+  // Get user details from localStorage and save to react store
+  try {
+    const info = localStorage.getItem("userInfo");
+    if (info) {
+      let userInfo = JSON.parse(info);
+      console.log("Loaded UserId from storage : " + userInfo.userId);
+      console.log("Loaded token from storage : " + userInfo.token);
+      props.setCurrentUser({
+        userId: userInfo.userId,
+        token: userInfo.token,
+      });
+    } else {
+      console.log("user info not found");
     }
+  } catch {
+    console.log("Cannot find info from localStorage");
   }
+}
 
-  onMenuToggle() {
-    const overlay = this.state.overlay;
-    this.setState({
-      overlay: !overlay,
-    });
-  }
+function App(props) {
+  const [overlay, setOverlay] = useState(false);
 
-  render() {
-    return (
-      <div>
-        <MainContainer>
-          <NavigationBar onMenuToggle={this.onMenuToggle} menuToggled={this.state.overlay} />
-          <PageContainer>
-            <Sidebar overlay={this.state.overlay} />
-            {this.state.overlay && <Overlay onClick={this.onMenuToggle} />}
-            <Page>
-              <Route render={({ location }) => {
-                return <TransitionGroup>
-                  <Transition
-                    key={location.key}
-                    timeout={300}
-                    classNames="page-fade"
-                  >
-                    <div>
-                      <Switch location={location}>
-                        <Route exact path="/" component={this.props.currentUser ? HomepageLoggedIn : HomePage} />
-
-                        <Route exact path="/signin" component={() => <Redirect to="/login" />} />
-                        <Route exact path="/login" component={() => <SignInAndSignUpPage inputBox={<SignIn />} />} />
-                        <Route exact path="/signup" component={() => <SignInAndSignUpPage inputBox={<SignUp />} />} />
-
-                        <Route exact path="/communities" component={CommunitiesPage} />
-                        <Route exact path="/community/new" component={CreateCommunityPage} />
-                        <Route exact path="/community/:uuid" component={CommunityPage} />
-
-                        <Route exact path="/user/:uuid" component={UserPage} />
-                        <Route exact path="/house/:uuid" component={HousePage} />
-
-                        <Route path="*" component={NotFoundPage} />
-                      </Switch>
-                    </div>
-                  </Transition>
-                </TransitionGroup>
-              }} />
-            </Page>
-          </PageContainer>
-        </MainContainer>
-      </div>
-    );
+  useEffect(() => {
+    loadUserInfo(props);
+  });
```

Comment:
This function is as far as I can tell called in a loop.

Maybe we can use the `useCallback` hook here?

Reply:
How do you figure it's called in a loop? From what I read, `useEffect` is the equivalent of `componentDidMount`, so it'll fire one time when the component mounts.

Reply:
I looked in the console, and it loaded the  user data hundreds times ever repeating.
I'd figure it has something to do with that we change the props and then the component re-renders, not entirely sure about that though.
I stumbled upon a similar problem in [`HomepageLoggedIn.jsx#L125L132`](https://github.com/jmprathab/MyHome-Web/blob/master/src/pages/homepage/HomepageLoggedIn.jsx#L125L132), no gurantee that it will help with this one here too though

Reply:
@DZwell @Tobi406 is right, take a look here:
https://dmitripavlutin.com/react-useeffect-infinite-loop/

Reply:
maybe you should add an empty list to resolve to emulate the `componentDidMount` behavior
```
 useEffect(() => {
    loadUserInfo(props);
  }, []);
```